include/gatekeeper_sol.h

```diff
@@ -0,0 +1,131 @@
+/*
+ * Gatekeeper - DoS protection system.
+ * Copyright (C) 2016 Digirati LTDA.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef _GATEKEEPER_SOL_H_
+#define _GATEKEEPER_SOL_H_
+
+#include <rte_approx.h>
+#include <rte_cycles.h>
+#include <rte_reciprocal.h>
+
+#include "list.h"
+
+struct priority_req {
+	/* Doubly-linked list node. */
+	struct list_head    list;
+	/* The packet for this request. */
+	struct rte_mbuf     *pkt;
+	/* The priority of this request. */
+	uint8_t             priority;
+};
+
+/* Number of priorities available in the IP DSCP field. */
+#define GK_NUM_REQ_PRIORITIES (64)
```

Comment:
Almost everywhere `GK_NUM_REQ_PRIORITIES` is being used, it's meant to mean 63 instead of 64. I suggest renaming this constant to `GK_MAX_REQ_PRIORITY` and adding plus one where needed. Don't forget to adjust the comment.